tiledb/sm/array_schema/enumeration.cc

```diff
@@ -235,14 +243,24 @@ shared_ptr<const Enumeration> Enumeration::extend(
     uint64_t data_size,
     const void* offsets,
     uint64_t offsets_size) const {
-  if (data == nullptr) {
-    throw EnumerationException(
-        "Unable to extend an enumeration without a data buffer.");
-  }
+  if (data == nullptr && data_size == 0) {
+    // This is OK
+    //
+    // Users need to be able to create an enumeration containing
+    // one value which is the empty string.
+    //
+    // The std::vector<uint8_t> foo(0, 0) constructor at our callsite results
+    // in a foo.data() which is nullptr and foo.size() which is zero.
+  } else {
+    if (data == nullptr) {
+      throw EnumerationException(
+          "Unable to extend an enumeration without a data buffer.");
+    }
 
-  if (data_size == 0) {
-    throw EnumerationException(
-        "Unable to extend an enumeration with a zero sized data buffer.");
+    if (data_size == 0) {
+      throw EnumerationException(
+          "Unable to extend an enumeration with a zero sized data buffer.");
+    }
   }
```

Comment:
An enumeration with a single empty string is represented by an empty data buffer and a single offset of zero. It would be weird if the constructor accepted a `{nullptr, 0}` data buffer but the `extend` method didn't.

Reply:
They both do,  as unit-tested on this PR.

Reply:
Oops I completely missed it. Never mind. 😅